### ISSUE TITLE
core tests do not run under wasm32-unknown-unknown

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#92841]
I was looking at the CI for rust-lang/rust running wasm32-unknown-unknown tests and I noticed that the library/core tests were being compiled, but would instantly finish running without an error.

I would expect that if there is a problem, it should return an error, or it should run the core tests (which I would expect should take more than 0.5 seconds to run).

I tried running a debugger with the `.wasm` file, but it seem to initially launch a thread, and I don't see how to debug that.

Testing the same compiler with a trivial `#[test] fn foobar() { assert_eq!(1,2); }` confirms that tests *should* be working (that example fails as expected).

cc @alexcrichton not sure if you would be interested in this.

### Meta

Using a recent master: 72e74d7b9cf1a7901650227e74650f1fcc797600

**Comment by alexcrichton** [comment]
I think this may be a case where the tests are just fast? With a lack of output I agree that it's hard to figure out, and my guess is that any long-running "interesting" tests are probably ignored on wasm for portability reasons (e.g. no threads, no system resources, etc).

I changed a random test tin `library/core/tests/*.rs` to fail and I indeed got a failure running `./x.py --stage 2 test --host='' --target=wasm32-unknown-unknown library/core`, so I think that tests may be running?

Now that being said I think it would probably be better to switch to wasi at some point for tests since it makes them at all debuggable.
